lxd/oidc_sessions.go

```diff
@@ -0,0 +1,319 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+
+	"github.com/canonical/lxd/lxd/auth"
+	"github.com/canonical/lxd/lxd/db"
+	dbCluster "github.com/canonical/lxd/lxd/db/cluster"
+	"github.com/canonical/lxd/lxd/request"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/lxd/util"
+	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/entity"
+	"github.com/canonical/lxd/shared/version"
+)
+
+var oidcSessionsCmd = APIEndpoint{
+	Name:        "oidc_sessions",
+	Path:        "auth/oidc-sessions",
+	MetricsType: entity.TypeIdentity,
+	Get: APIEndpointAction{
+		Handler:       oidcSessionsGet,
+		AccessHandler: allowAuthenticated,
+	},
+}
+
+var oidcSessionCmd = APIEndpoint{
+	Name:        "oidc_session",
+	Path:        "auth/oidc-sessions/{id}",
+	MetricsType: entity.TypeIdentity,
+	Get: APIEndpointAction{
+		Handler: oidcSessionGet,
+		// Caller can view the session if they can view the identity.
+		// All identities can view themselves.
+		AccessHandler: oidcSessionAccessHandler(auth.EntitlementCanView),
+	},
+	Delete: APIEndpointAction{
+		Handler: oidcSessionDelete,
+		// Caller can delete the session if they can delete the identity.
+		// All identities can delete themselves.
+		AccessHandler: oidcSessionAccessHandler(auth.EntitlementCanDelete),
+	},
+}
+
+const ctxOIDCSessionDetails request.CtxKey = "session-details"
+
+func oidcSessionAccessHandler(entitlement auth.Entitlement) func(d *Daemon, r *http.Request) response.Response {
+	return func(d *Daemon, r *http.Request) response.Response {
+		s := d.State()
+		sessionIDStr, err := url.PathUnescape(mux.Vars(r)["id"])
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		sessionID, err := uuid.Parse(sessionIDStr)
+		if err != nil {
+			return response.BadRequest(fmt.Errorf("Bad session ID: %w", err))
+		}
+
+		var session *dbCluster.OIDCSession
+		err = s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+			session, err = dbCluster.GetOIDCSessionByUUID(ctx, tx.Tx(), sessionID)
+			return err
+		})
+		if err != nil {
+			if api.StatusErrorCheck(err, http.StatusNotFound) {
```

Comment:
why not response.SmartError here so that the DB level "Session not found" bubbles up rather than generic not found error?